nss-day-cohort-52/nashville-kennels-pine-bears
# Non-employees can now successfully register a new animal

### Issue ticket(s): #11 

### New file(s): `AnimalCaretakerRepository.js`

### Modified File(s): `AnimalForm.js`

### Modifications:
1.AnimalForm.js
- Added a `locations `state variable on line 12
- Added another `useEffect ` on line 21 to set the `locations `state variable
- Added a `findLocation()` function on line 32 whose responsibility is to return a single location object based on the conditions set in the function
- Modified the `.then`'s that follow the `addAnimal` function on line 50 to post an animalCaretaker object to the API, the new animalCaretaker object is dependent on the object that we passed into the `addAnimal` function above it
2. AnimalCaretakerRepository.js
- Just copied over the code from the `AnimalOwnerRepository.js` and refactored it to affect the AnimalCaretakers table 
 
### Steps to test:
1. While in the terminal, use git fetch --all in the root directory of this repo
2. Checkout out to `al-registerAnimal`
3. Log in to the application as a user who is NOT an employee
4. Navigate to the AnimalList component by selecting the "Animals" link from the navbar
5. click on the "Register Animal" button at the top of the page
6. Fill out the form in it's entirety
7. Before you click "Submit" open devtools and navigate to the Network tab
8. Click "Submit" and you should see two new objects show up in the devtools. One should be an Animal object, and the other should be an AnimalCaretaker object. Confirm that all of the foreign keys are being related correctly.
9. Finally, you should automatically be sent back to the AnimalList component where your new Animal object will be displayed as a card in the list. Expand the card and you should see the Caretaker that you selected in the form displayed
